Source/LeapMotion/Private/LeapWrapper.cpp

```diff
@@ -315,32 +345,50 @@ void FLeapWrapper::HandleDeviceEvent(const LEAP_DEVICE_EVENT *DeviceEvent)
 			return;
 		}
 	}
+
 	SetDevice(&DeviceProperties);
 
+	/*Result = LeapSubscribeEvents(ConnectionHandle, DeviceHandle);
+	if (Result != eLeapRS_Success)
+	{
+		UE_LOG(LeapMotionLog, Warning, TEXT("failed to subscribe to device %s.\n"), ResultString(Result));
+		return;
+	}*/
+
+	//Link device_id and handle
+	//DeviceHandles.Add(DeviceEvent->device.id, DeviceHandle);
+
 	if (CallbackDelegate) 
 	{
 		TaskRefDeviceFound = FLeapAsync::RunShortLambdaOnGameThread([DeviceEvent, DeviceProperties, this]
 		{
 			if (CallbackDelegate)
 			{
 				CallbackDelegate->OnDeviceFound(&DeviceProperties);
+				free(DeviceProperties.serial);
 			}
 		});
 	}
+	else {
+		free(DeviceProperties.serial);
+	}
 
-	free(DeviceProperties.serial);
-	LeapCloseDevice(DeviceHandle);
+	//LeapCloseDevice(DeviceHandle);
```

Comment:
@lazloringUH are you happy to removed the commented out code or does it require more work?